libs/payments/ui/src/lib/server/components/PurchaseDetails/index.tsx

```diff
@@ -57,16 +58,28 @@ type PurchaseDetailsProps = {
     productName: string;
     webIcon: string;
   };
+  discountEnd?: number | null;
```

Comment:
nit: this type is `number | null | undefined` - do we want both `null` and `undefined as viable types here?

Reply:
Ah, I see above that this can be `null` if the coupon is repeating without end. Correct? If so, please ignore :)

Reply:
Good question. A `discountEnd` could also be `null` if the discount is `forever`, and it would be `undefined` if there is no discount, so I think both are viable types. 👍🏼